week-13/homework2/index2.js

```diff
@@ -0,0 +1,208 @@
+    currentQuestionIndex = pageIndex;
+    //set all links to inactive
+    let allBtns = nav.querySelectorAll('.page-item');
+    for (let i = 0; i < allBtns.length; i++) {
+        allBtns[i].classList.remove('active');
+    }
+    renderPage();
+}
+
+let renderPage = function () {
+    let currentQuestion = quizData[currentQuestionIndex];
+    let question = currentQuestion.question;
+    let answers = currentQuestion.answers;
+
+    //Display first question
+    questionElement.innerHTML = `<span class="fw-bold">Question ${currentQuestionIndex + 1}.</span> ${question}`;
+
+    //clear buttons
+    answerElement.innerHTML = '';
+
+    //loop through answers and add button
+    for (let i = 0; i < answers.length; i++) {
+        let button = document.createElement('button');
+        button.textContent = answers[i].text;
+        button.classList.add('btn', 'btn-outline-secondary', 'd-block');
+        //add event listener to button
+        button.addEventListener('click', function () {
+            checkAnswer(button, answers[i]);
+        })
+        answerElement.appendChild(button);
+    }
+
+    //sets nav element to blank and render the navigation
+    nav.innerHTML = '';
+    for (let i = 0; i < quizData.length; i++) {
+        let newPageNum = document.createElement('li');
+        newPageNum.classList.add('page-item');
+        let newPageLink = document.createElement('a');
+        newPageLink.innerHTML = i + 1;
+        newPageLink.classList.add('page-link');
+        newPageLink.addEventListener('click', function () {
+            onNavClick(i);
+        });
+        newPageNum.appendChild(newPageLink);
+        nav.appendChild(newPageNum);
+    }
+
+    //sets the active page number
+    let allBtns = nav.querySelectorAll('.page-item');
+    allBtns[currentQuestionIndex].classList.add('active');
+
+    //checks to see if question has been answered
+    if (selectedAnswers[currentQuestionIndex]) {
+        //select all btns
+        let allBtns = answerElement.querySelectorAll('.btn');
+        let currentBtn;
+        //check to see if its the current btn
+        for (let i = 0; i < allBtns.length; i++) {
+            if (allBtns[i].innerHTML == selectedAnswers[currentQuestionIndex].text) {
+                currentBtn = allBtns[i];
+            }
+        }
+        checkAnswer(currentBtn, selectedAnswers[currentQuestionIndex]);
+    }
+
+    //checks whether or not to display submit btn
+    if (currentQuestionIndex == quizData.length - 1) {
+        submitBtn.classList.remove('d-none');
+        nextBtn.classList.add('d-none');
+    } else {
+        nextBtn.classList.remove('d-none');
+        submitBtn.classList.add('d-none');
+    }
+
+}
+
+//When next btn is pressed get next question
+let nextQuestion = function () {
+    currentQuestionIndex++;
+    if (currentQuestionIndex < quizData.length) {
+        renderPage();
+    } else {
+        container.innerHTML = '';
+        const hours = Math.floor(time / 3600);
```

Comment:
You should move this block https://github.com/TaiChiLe/fullstack-learning/pull/9/files#diff-c62c9007d77bb7665156ea99de3a66b1ff8208d60b48db185e6f9136dd3fd1c2R168-R171
to a different method outside so you will not need to duplicate the code to format time to string in 2 places (here and in the updatedTimer function)